### PR TITLE
inductor: promote half/bfloat16 constant to float for cpu vectorization path

### PR DESCRIPTION
**Pull request by XiaobingSuper**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #105440

As scalar path, we should also promote half/bfloat16 constant to float for better accuracy, after this PR, the TIMM ```dm_nfnet``` model amp path can be passed.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

### CODE REVIEW

File: torch/_inductor/codegen/cpp.py [hunk 1]
```diff
@@ -2092,6 +2092,10 @@ def store_reduction(name, index, value):
 
             @staticmethod
             def constant(val, dtype):
+                if dtype in (torch.float16, torch.bfloat16):
```
Comment (jgong5) [inline torch/_inductor/codegen/cpp.py:2095]: we are not checking float16 here: https://github.com/pytorch/pytorch/pull/105440/files#diff-5ab7b0235e2076a5fc6629ba0b109208940f5b94f5c13babc3e0f87cf4fcec82R2120. Can float16 work well in the test you added?

File: test/inductor/test_cpu_repro.py [hunk 2]
```diff
@@ -1968,6 +1968,14 @@ def f(a):
         x = torch.rand(16)
         self.common(f, (x,))
 
+    def test_scalar_mul_low_precison(self):
+        def f(x):
+            return torch.ops.aten.mul.Tensor(x, 1.7015043497085571)
+
+        for dtype in [torch.float16, torch.bfloat16]:
+            x = torch.randn(4, 5, dtype=dtype)
+            self.common(f, (x,))
```
Comment (jgong5) [inline test/inductor/test_cpu_repro.py:1977]: add a check for the count of generated vec kernel?

**Review by jgong5 — CHANGES_REQUESTED** [review]

File: torch/_inductor/codegen/cpp.py (hunk 1 above)
Comment (XiaobingSuper) [reply torch/_inductor/codegen/cpp.py:2095]: yes， it can works for my test.

File: test/inductor/test_cpu_repro.py (hunk 2 above)
Comment (XiaobingSuper) [reply test/inductor/test_cpu_repro.py]: added.

**Review by jgong5 — APPROVED** [review]

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**